notebooks/Time_Series_Forecasting.ipynb

```diff
@@ -0,0 +1,660 @@
+    "train_data.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_data_copy = train_data.copy()\n",
+    "train_data_copy.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plt.figure(figsize=(20, 100))\n",
+    "for i, family_value in zip(range(len(family_values)), family_values):\n",
+    "  plt.subplot(17, 2, i+1)\n",
+    "  for city in train_data_copy['city'].unique():\n",
+    "      average_sales = train_data_copy[(train_data_copy['family'] == family_value) &\n",
+    "                                 (train_data_copy['city'] == city)].groupby('date').mean()['sales']\n",
+    "      trend = average_sales.rolling(\n",
+    "          window=365,\n",
+    "          center=True,\n",
+    "          min_periods=183,\n",
+    "      ).mean()\n",
+    "      ax = average_sales.plot(alpha=0)\n",
+    "      ax = trend.plot(ax=ax, linewidth=3)\n",
+    "  plt.title(family_value)\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plt.figure(figsize=(7,5))\n",
+    "sns.countplot(x=train_data['type'], alpha=0.7, data=train_data)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "plt.figure(figsize=(20, 100))\n",
+    "for i, family_value in zip(range(len(family_values)), family_values):\n",
+    "  plt.subplot(17, 2, i+1)\n",
+    "  for store_type in train_data_copy['type'].unique():\n",
+    "      #plt.plot(np.arange(len(current_type_and_family_data.index)), current_type_and_family_data['sales'])\n",
+    "      average_sales = train_data_copy[(train_data_copy['family'] == family_value) &\n",
+    "                                 (train_data_copy['type'] == store_type)].groupby('date').mean()['sales']\n",
+    "      trend = average_sales.rolling(\n",
+    "          window=365,\n",
+    "          center=True,\n",
+    "          min_periods=183,\n",
+    "      ).mean()\n",
+    "      ax = average_sales.plot(alpha=0)\n",
+    "      ax = trend.plot(ax=ax, linewidth=3)\n",
+    "  plt.title(family_value)\n",
+    "  plt.legend(train_data_copy['type'].unique())\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<i>As we can see, the store type has a great impact on the target (type D is leader and it is logical, because the plot above shows us, that type D stores are the most), that is why 'type' feature is very important.</i>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "plt.figure(figsize=(7,5))\n",
+    "sns.countplot(x=train_data['cluster'], alpha=0.7, data=train_data)"
```

Comment:
Nevertheless it's interesting to know the meaning of this `cluster` variable. Probably it is related to geography? Or store type?
If it's too hard, this question can be postponed for better times.